### ISSUE TITLE
defaultValue is broken

### ISSUE THREAD

**Issue by medihack** [react-monaco-editor/react-monaco-editor#674]
**Describe the bug**
Setting `defaultValue` is currently broken.

**To Reproduce**
It is even broken in the provided example. Take a look at the `Another editor (uncontrolled mode)`. It is empty even when just setting `defaultValue` to a static string.

**Expected behavior**
`defaultValue` resp. uncontrolled mode should work as expected.

**Environment (please complete the following information):**
react-monaco-editor v0.50.1
monaco-editor v0.34.1

- [ ] I will try to send a pull request to fix this issue.

**Comment by domoritz** [comment]
Did this break in a particular version?

**Comment by medihack** [comment]
With v0.49.0 it seemed to be ok., but I can't say for sure as I just switched from controlled to uncontrolled mode.

**Comment by medihack** [comment]
I guess the `defaultValue` is somehow overwritten by value (even if not set), because in `editorDidMount` the value is still correctly set with `defaultValue`, but after some milliseconds it is cleared. Maybe a problem with the `useEffect` of setting the value, but just a wild guess.

**Comment by domoritz** [comment]
Could the issue be #625?

**Comment by medihack** [comment]
Yes, I think so. From having a quick look at the code I think [this `useEffect`](https://github.com/react-monaco-editor/react-monaco-editor/blob/2d2fe2ed2c19021841c558bcbea68113836198a4/src/editor.tsx#L85) is wrong as the value is always set even when `value` was not provided. And thereby it overwrites the value set by `defaultValue`. I guess a simple way to fix this is to simply use `if (editor.current && value !== null) {` [here](https://github.com/react-monaco-editor/react-monaco-editor/blob/2d2fe2ed2c19021841c558bcbea68113836198a4/src/editor.tsx#LL86C5-L86C26). Unfortunately, I don't have the time to test this and come up with a PR. But it could be the problem.
